src/MergeContext.js

```diff
@@ -984,7 +980,7 @@ class PullRequest {
         }
 
         if (!(await this._stagedCommitIsFresh())) {
-            await this._labels.addImmediately(Config.abandonedStagingChecksLabel());
+            this._signalAbandonmentOfStagingChecks = true;
```

Comment:
This change cancels the idea from 5ef56f6, i.e., to reflect the 'abandoning' fact in the PR history. Now there will be no abandonedStagingChecksLabel() if a staged commit becomes stale (and abandoned) and another staged commit is immediately created. Is this the intention? I could not get this from the [detailed] 3394c6c description.

Reply:
> Now there will be no abandonedStagingChecksLabel() if a staged commit becomes stale (and abandoned) and another staged commit is immediately created.

Agreed. I forgot that we apparently wanted that signal. On one hand, it feels like the creation of another staged commit is enough of a clue on its own, but perhaps that clue gets obscured in the other noise somehow. I suggest to get rid of that signal as an experiment. If we realize it was actually useful, then we will restore it (hopefully without recreating the bug this PR is fixing). AFAICT, the difficult parts of this PR will still be needed even if we decide to resurrect that signal later.

Reply:
I adjusted the PR description to reflect the current code and the above suggestion, but we can still refactor this in this PR if we are sure that short-lived signal should be kept.

Reply:
> get rid of that signal as an experiment.

Ok, let's try to keep things simple for now. I am fine with this PR and we can merge it then.